codalab/lib/codalab_manager.py

```diff
@@ -555,15 +558,22 @@ def check_version(self, server_version):
 
     def logout(self, address):
         """Clear credentials associated with given address."""
-        if address in self.state['auth']:
-            del self.state['auth'][address]
+        state = self.get_state()
+        if address in state['auth']:
+            del state['auth'][address]
             self.save_state()
 
     def save_config(self):
         if self.temporary:
             return
         write_pretty_json(self.config, self.config_path)
 
+    def get_state(self):
```

Comment:
This isn't absolutely necessary for this PR, but it might be cleaner (and maybe be good as a future improvement to improve readability / reduce mistakes) to just make `this.state` a property, then define getters and setters for it as shown here: https://docs.python.org/3/library/functions.html#property